core/src/main/java/com/coderanch/blackjack/Card.java

```diff
@@ -13,48 +13,46 @@
  * A playing card used in a game of Blackjack.
  */
 final class Card {
-  
+
   /**
-  * The rank of {@link} card
-  **/
-  enum Rank { ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING };
-  
+   * The rank of {@link} card.
+   */
+  enum Rank { ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING }
+
+
   /**
-  * The suit of {@link} card.
-  **/
-  enum Suit { SPADES, HEARTS, CLUBS, DIAMONDS };
-  
+   * The suit of {@link} card.
+   */
+  enum Suit { SPADES, HEARTS, CLUBS, DIAMONDS }
+
   private final Rank rank;
   private final Suit suit;
-  
+
   /**
-  * Constructor.
-  * @param rank (required) Rank value. Must have value of enum Rank.
-  * @param suit (required) Suit value. Must have value of enum Suit.
-  */
-  
+   * Constructor.
+   *
+   * @param rank (required) Rank value. Must have value of enum Rank.
+   * @param suit (required) Suit value. Must have value of enum Suit.
+   */
   public Card(Rank rank, Suit suit) {
-    if (rank == null || suit == null) {
-      throw new IllegalArgumentException("Rank and suit should not be null");
-    }
-    this.rank = rank;
-    this.suit = suit;
+    this.rank = Objects.requireNonNull(rank, "Rank should not be null");
+    this.suit = Objects.requireNonNull(suit, "Suit should not be null");
```

Comment:
This change causes the tests to fail because they expect `IllegalArgumentException`. We need to decide what exception type we want to use for null-checks in method/constructor parameters. If we do want to stick with `IllegalArgumentException`, we could include my Require library to make this process less painful.

Reply:
For now, I suggest we change the tests to expect NullPointerException, we can always change it later.

Reply:
Sorry, I wasn't aware about such existing test. I need to get more familiar with where this project headed to already. I was researching some time ago which exception is more appropriate to throw when unwanted null is passed as a parameter: either NullPointer or IllegalArgument exception, and came to conclusion, that IllegalArgumentException is more appropriate when there is non-null unwanted parameter value, and NullPointerException when there is actually a null, so you know right away what you are dealing with. What is your opinion about that?

Reply:
I prefer the option that gives more fine grained information about what went wrong. A `NullPointerException` to me implies that you dereferenced null, not necessarily because you supplied a wrong argument to a method. Throwing `NullPointerException` without actually dereferencing the value hides that the problem was caused because you passed a wrong argument. I find that I care that a wrong argument was passed, not necessarily that that argument was null or something else.

I don't have a very strong preference, but I find it easier to debug an application when using `IllegalArgumentException`.